### ISSUE TITLE
called `Option::unwrap()` on a `None` value

### ISSUE THREAD

**Issue by jlkravitz** [prisma/prisma#21219]
Hi Prisma Team! My Prisma Client just crashed. Perhaps related to #10586?

This is coming up repeatedly and really making it difficult to work with prisma...

This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v18.15.0           | 
| OS              | darwin-arm64       |
| Prisma Client   | 4.16.2             |
| Query Engine    | 4bc8b6e1b66cb932731fb1bdbbc550d1e010de81|
| Database        | sqlserver          |

## Logs
```
prisma:tryLoadEnv Environment variables loaded from /Users/joshua/Documents/matchmaker/backend/.env
prisma:client checkPlatformCaching:postinstall false
prisma:client checkPlatformCaching:ciName 
prisma:tryLoadEnv Environment variables loaded from /Users/joshua/Documents/matchmaker/backend/.env
prisma:client dirname /Users/joshua/Documents/matchmaker/backend/node_modules/.prisma/client
prisma:client relativePath ../../../prisma
prisma:client cwd /Users/joshua/Documents/matchmaker/backend/prisma
prisma:client protocol graphql
prisma:client clientVersion 4.16.2
prisma:client clientEngineType library
prisma:client:libraryEngine internalSetup
prisma:client:engines:resolveEnginePath enginePath /Users/joshua/Documents/matchmaker/backend/node_modules/.prisma/client/libquery_engine-darwin-arm64.dylib.node
prisma:client:libraryEngine library starting
prisma:client:libraryEngine library started
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:libraryEngine sending request, this.libraryStarted: true
prisma:client:request_handler {"name":"PrismaClientRustPanicError","clientVersion":"4.16.2"}
prisma:client:libraryEngine sending request, this.libraryStarted: true
pri

**Comment by aqrln** [comment]
@jlkravitz thanks for the report, could you please provide the missing information so we could reproduce the issue (the query that is failing and the schema)?

**Comment by jlkravitz** [comment]
> @jlkravitz thanks for the report, could you please provide the missing information so we could reproduce the issue (the query that is failing and the schema)?

@aqrln Just added a minimal example based on my code that produces the error.

**Comment by janpio** [comment]
@jlkravitz That makes a lot of sense. Optimally Prisma would of course not crash here, but return some useful error message to the user what went wrong - according to your description it queried for 1 entry (as indicated by the relation and `@unique`) but got back multiple ones. Seems we currently just do not have any code to handle that case.

**But** I have trouble to reproduce the situation you are describing. When I try to manually create a schema similar to the one above, 2 tables with a relation to a _non-unique_ field, with this SQL:
```sql
-- CreateTable
CREATE TABLE [dbo].[actions] (
    [id] INT NOT NULL IDENTITY(1,1),
    [request_id] INT NOT NULL,
    CONSTRAINT [actions_pkey] PRIMARY KEY CLUSTERED ([id])
);

-- CreateTable
CREATE TABLE [dbo].[request2] (
    [id] INT NOT NULL IDENTITY(1,1),
    [request_id] INT NOT NULL,
    CONSTRAINT [request2_pkey] PRIMARY KEY CLUSTERED ([id]),
    -- CONSTRAINT [request2_request_id_key] UNIQUE NONCLUSTERED ([request_id])
);

-- AddForeignKey
ALTER TABLE [dbo].[actions] ADD CONSTRAINT [actions_request_id_fkey] FOREIGN KEY ([request_id]) REFERENCES [dbo].[request2]([request_id]) ON DELETE NO ACTION ON UPDATE CASCADE;

```
(Note the commented out `UNIQUE` constraint)

I get an error message from SQL server on creation:
```
There are no primary or candidate keys in the referenced table 'dbo.request2' that match the referencing column list in the foreign key 'actions_request_id_fkey'.
```

Seems SQL Server does not allow you to create a schema with a foreign key (to represent the relation) that points to a non unique column?

Is this maybe where the views come in?
Or does your table not actually have a foreign key, but you only added that to the Prisma schema?

**Comment by janpio** [comment]
Does that mean the `@relation` was also not really backed by a foreign key, but that it only existed in the Prisma schema so that the API is generated to use it like one? Then we could indeed end up in a situation that could work like yours.

**Comment by jlkravitz** [comment]
Yes- isn't that by definition the case since this is a view? AFAIK, I
couldn't set up a foreign key onto a view, right?

**Comment by jlkravitz** [comment]
Hi @laplab Makes complete sense. I will emphasize, though, that the lack of specificity in the error message made this an _extremely_ difficult thing to debug. Even if it had said that "there is a mismatch between the database and prisma schemas," that may have saved several hours. I have encountered this same error message in different circumstances, too. It's quite confusing when a low-level error (in Rust) appears on what is supposed to be a high-level tool (in Typescript).

More than anything, I guess this is my pitch for fixing the error message, even if the use case is (understandably) not supported. If you'd like me to include a ticket or link this one to an existing one, let me know. I'm happy to help, but am not a regular contributor to this project so am not familiar with its processes.
